pkg/controller/pxc/status.go

```diff
@@ -124,6 +127,15 @@ func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluste
 		})
 	}
 
+	err = r.setUpgradeInProgress(cr)
+	if err != nil {
+		log.Info(err.Error())
+	}
+
+	if cr.Status.UpgradeInProgress {
+		cr.Status.Status = api.AppStateInit
+	}
```

Comment:
All `Status` assignments are done in the `switch` above. I don't see any necessity to split this logic and introduce additional `if` here